### DISCUSSION TITLE
Request failed Google Pay live mode

### DISCUSSION THREAD

**Discussion by panchalAlpit** [flutter-stripe/flutter_stripe#843] [Q&A]
Hello All,
Google pay is completely working on Test mode but after set the live secret key in "v1/payment_intents"  in this API so it's not working and getting the below error.

![WhatsApp Image 2022-07-20 at 5 22 21 PM](https://user-images.githubusercontent.com/48007084/179978810-0db30f55-4502-4ca3-8a54-ffafae6d5829.jpeg)

     `.  Future<void> startGooglePay() async {
            final googlePaySupported = await Stripe.instance.isGooglePaySupported(IsGooglePaySupportedParams(testEnv: 
            false));
             if (googlePaySupported) {
            try {
               // 1. fetch Intent Client Secret from backend
             final response = await fetchPaymentIntentClientSecret();
           print(response);
               final clientSecret = response['client_secret'];

         // 2.present google pay sheet

        await Stripe.instance.initGooglePay(GooglePayInitParams( merchantName: "Test Name",
            countryCode: 'US',testEnv: false));

        await Stripe.instance.presentGooglePay(PresentGooglePayParams(clientSecret: clientSecret,currencyCode: "GBP"),);

        await Stripe.instance.initPaymentSheet(paymentSheetParameters: SetupPaymentSheetParameters(
            googlePay: PaymentSheetGooglePay(merchantCountryCode: 'US'),
            paymentIntentClientSecret: response['client_secret'],
            setupIntentClientSecret:response['client_secret'],
            customerEphemeralKeySecret: response['client_secret'],
            customerId:PLPrefrence.getPrefValue(key: PLPrefrence.user_code).toString()
        ));

        await Stripe.instance.retrievePaymentIntent(response['client_secret']).then((value){

          print(value.id);
          print(value.status);
          print(value.confirmationMethod.name);

        });

        // ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text('Google Pay payment succesfully completed')),);

      } catch (e) {
        print('ErrorStripePrint : ${e}');
        // ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text('ErrorStripe : $e')),);
        errorDialog(errorMessage: "$e");
      }
    } else {
      errorDialog(errorMessage: "Google pay is not supported on this device");
      // ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text('Google pay is not supported on this 
      device')),);
           }
       }`

Now here is the API call for payment_intents API

        `         Future<Map<String, dynamic>> fetchPaymentIntentClientSecret() async {
           final url = Uri.parse('https://api.stripe.com/v1/payment_intents');
           Map<String, dynamic> body = {
             'amount':calculateAmount(amount:  _plController.isUseWalletbalance.value ? 
          _plController.processedAmount.value : widget.grandTotal.toString()).toString(),
           'currency': "GBP",
           'payment_method_types[]': 'card',

           };

           final response = await http.post(
             url,
             headers: {
              'Content-Type': 'application/x-www-form-urlencoded',
              'Authorization': 'Bearer ${PLKeys.stripeSK_Live}',
            },
            body: body,
           );

        return json.decode(response.body);
       }`

if I set the testEnv true and test secret key in so it's working but set testEnv false and set stripe live mode secret key so it's not working and getting issue as upload image above.

**Comment by panchalAlpit** [comment]
Please help me for this.

**Comment by panchalAlpit** [comment]
Hello 
I have got the approval from [Google pay](https://prnt.sc/9wknuMREsxdi)  but still getting the same error as I attach the screenshot. So why I'm getting still this error in release android build. Please help me

**Comment by tankadesign** [comment]
Has you ever gotten Google Pay to work in flutter in production? Same issue here.
